src/foundry/client/data/documents/combat.d.ts

```diff
@@ -1,198 +1,210 @@
+
+    /**
+     * Loads the registered Combat Theme (if any) and plays the requested type of sound.
+     * If multiple exist for that type, one is chosen at random.
+     * @param type  - One of [ "startEncounter", "nextUp", "yourTurn" ]
+     * @internal
+     */
+    protected _playCombatSound(type: string): void;
 
     /**
      * Define how the array of Combatants is sorted in the displayed list of the tracker.
      * This method can be overridden by a system or module which needs to display combatants in an alternative order.
-     * By default sort by initiative, next falling back to name, lastly tie-breaking by combatant id.
+     * The default sorting rules sort in descending order of initiative using combatant IDs for tiebreakers.
+     * @param a   - Some combatant
+     * @param b   - Some other combatant
      * @internal
      */
-    protected _sortCombatants(
-      a: InstanceType<ConfiguredDocumentClass<typeof Combatant>>,
-      b: InstanceType<ConfiguredDocumentClass<typeof Combatant>>
-    ): number;
+    protected _sortCombatants(a: Combatant, b: Combatant): number;
 
-    protected override _onCreate(
-      data: this["data"]["_source"],
-      options: DocumentModificationOptions,
-      userId: string
-    ): void;
+    /** @internal */
+    protected _onCreate(data: BaseCombat.ConstructorData, options: DocumentModificationOptions, userId: string): void;
 
-    protected override _onUpdate(
-      changed: DeepPartial<this["data"]["_source"]>,
+    /** @internal */
+    protected _onUpdate(
+      data: DeepPartial<BaseCombat.UpdateData>,
       options: DocumentModificationOptions,
       userId: string
     ): void;
 
-    protected override _onDelete(options: DocumentModificationOptions, userId: string): void;
+    /** @internal */
+    protected _onDelete(options: DocumentModificationOptions, userId: string): void;
 
-    protected override _onCreateEmbeddedDocuments(
+    /** @internal */
+    protected _onCreateEmbeddedDocuments(
       type: string,
       documents: InstanceType<ConfiguredDocumentClass<typeof Combatant>>[],
-      result: CombatantDataSource[],
+      result: CombatantData[],
       options: DocumentModificationOptions,
       userId: string
     ): void;
 
-    protected override _onUpdateEmbeddedDocuments(
+    /** @internal */
+    protected _onUpdateEmbeddedDocuments(
       embeddedName: string,
       documents: InstanceType<ConfiguredDocumentClass<typeof Combatant>>[],
-      result: CombatantDataSource[],
-      options: DocumentModificationOptions,
+      result: DeepPartial<Combatant>[],
+      options: DocumentModificationContext,
       userId: string
     ): void;
 
-    protected override _onDeleteEmbeddedDocuments(
+    /** @internal */
+    protected _onDeleteEmbeddedDocuments(
       embeddedName: string,
       documents: InstanceType<ConfiguredDocumentClass<typeof Combatant>>[],
-      result: string[],
+      result: DeepPartial<Combatant>[],
```

Comment:
Are you sure? For other documents it always seemed like it was the ID that was passed here. But even if, it's probably `BaseCombatant.Source`.